sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/mapper/GenericLdapMapper.java

```diff
@@ -257,71 +257,77 @@
           entity.getClass().getDeclaredField(ldapField.getAnnotation(MapToAttribute.class).value());
       entityField.setAccessible(true);
       Object attributeValue = entityField.get(entity);
-      String attributeName = ldapField.getAnnotation(AttributeLdapName.class).value();
-      if (attributeValue != null) {
-        ModelType type = ldapField.getAnnotation(MapToAttribute.class).type();
-        switch (type) {
-          case ORGANIZATION:
-            Attribute organizationAttribute =
-                new Attribute(
-                    attributeName,
-                    String.format(
-                        "%s=%s,%s",
-                        OrganizationLdap.class
-                            .getAnnotation(LdapObjectClass.class)
-                            .rdnAttributeName(),
-                        ((Organization) attributeValue).getIdentifiant(),
-                        config.get("organization_source")));
-            List<Attribute> organizationAttributeList = new ArrayList<>();
-            organizationAttributeList.add(organizationAttribute);
-            return organizationAttributeList;
-          case LIST_HABILITATION:
-            return ((List<Habilitation>) attributeValue)
-                .stream()
-                    .filter(
-                        habilitation ->
-                            habilitation.getApplication() != null && habilitation.getRole() != null)
-                    .map(habilitation -> new Attribute(attributeName, habilitation.getId()))
-                    .collect(Collectors.toList());
-          case LIST_USER:
-            return new ArrayList<>();
-          case LIST_GROUP:
-            return ((List<Group>) attributeValue)
-                .stream()
-                    .map(
-                        group ->
-                            new Attribute(
-                                attributeName,
-                                String.format(
-                                    GroupLdap.class
-                                        .getAnnotation(LdapObjectClass.class)
-                                        .rdnAttributeName(),
-                                    group.getName(),
-                                    config.get("appli_source"))))
-                    .collect(Collectors.toList());
-          case ADDRESS:
-            List<Attribute> addressAttributeList = new ArrayList<>();
-            if (((Map<String, String>) attributeValue).containsKey("id")
-                && config.get("address_source") != null) {
-              Attribute addressAttribute =
+      if (!ldapField.getAnnotation(MapToAttribute.class).readonly()) {
```

Comment:
wouldn't it be better to filter at the filter level line 95 before entering createAttributesFromAttributeAnnotation ?

Reply:
maybe